### ISSUE TITLE
append, prepend return wrong type

### ISSUE THREAD

**Issue by chochos** [ceylon/ceylon-compiler#375]
ceylon.language has some tests that should pass:

``` ceylon
    assert(append({},"foo").size==1, "append to empty.size");
    assert(prepend({},"foo").size==1, "prepend to empty.size");
    assert(append({1, 2},"foo").size==3, "append.size");
    assert(prepend({1, 2},"foo").size==3, "prepend.size");
    assert(append({"one", "two" , "three"}, "four").size==4, "append");
```

But they fail, with two weird errors: `unknown is not a subtype of Equality` and `member method or attribute does not exist: size in type Object`. I assume this is an error in the compiler because ceylon-js compiles these same tests and typechecker doesn't throw any errors like these, but it could be a bug in ceylon.language as well.
